pkg/controller/pd_control.go

```diff
@@ -32,6 +32,34 @@ func GetPDClient(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) p
 	return pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.IsTLSClusterEnabled())
 }
 
+// Retry to GetPDClient for multi-cluster
+// If the pdClient built from the PD service name is unavailable, try to
+// build another one with the ClientURL in the PeerMembers.
+// ClientURL example:
+// ClientURL: https://cluster2-pd-0.cluster2-pd-peer.pingcap.svc.cluster2.local
+func GetPDClient(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
+	pdClient := GetPDClientBasic(pdControl, tc)
+
+	if len(tc.Status.PD.PeerMembers) == 0 {
+		return pdClient
+	}
+
+	_, err := pdClient.GetHealth()
+	if err == nil {
+		return pdClient
+	}
+
+	for _, pdMember := range tc.Status.PD.PeerMembers {
+		pdClient = pdControl.GetPeerPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.IsTLSClusterEnabled(), pdMember.ClientURL, pdMember.Name)
+		_, err := pdClient.GetHealth()
+		if err == nil {
+			return pdClient
+		}
+	}
+
+	return pdClient
```

Comment:
If you plan to return the pdClient built in L41, then this has been changed in L53.